yt/visualization/plot_window.py

```diff
@@ -440,6 +451,101 @@ def set_origin(self, origin):
+        `_setup_plots()` will need to be called after this function
+        to make the axes available for annotation.
+
+        Parameters
+        ----------
+        mpl_proj : string, tuple
+            if passed as a string, mpl_proj is the specified projection type,
+            if passed as a tuple, then tuple will take the form of
+            ("ProjectionType", (args)) or ("ProjectionType", (args), {kwargs})
+            Valid projection type options include:
+                'PlateCarree', 'LambertConformal', 'LabmbertCylindrical',
+                'Mercator', 'Miller', 'Mollweide', 'Orthographic',
+                'Robinson', 'Stereographic', 'TransverseMercator',
+                'InterruptedGoodeHomolosine', 'RotatedPole', 'OGSB',
+                'EuroPP', 'Geostationary', 'Gnomonic', 'NorthPolarStereo',
+                'OSNI', 'SouthPolarStereo', 'AlbersEqualArea',
+                'AzimuthalEquidistant', 'Sinusoidal', 'UTM',
+                'NearsidePerspective', 'LambertAzimuthalEqualArea'
+
+
+        Examples
+        --------
+
+        This will create a Mollweide projection using Mollweide default values
+        and annotate it with coastlines
+
+        >>> import yt
+        >>> ds = yt.load("")
+        >>> p = yt.SlicePlot(ds, "altitude", "AIRDENS")
+        >>> p.set_mpl_projection('AIRDENS', 'Mollweide')
+        >>> p._setup_plots()
+        >>> p.plots['AIRDENS'].axes.coastlines()
+        >>> p.show()
+
+        This will move the PlateCarree central longitude to 90 degrees and
+        annotate with coastlines.
+
+        >>> import yt
+        >>> ds = yt.load("")
+        >>> p = yt.SlicePlot(ds, "altitude", "AIRDENS")
+        >>> p.set_mpl_projection('AIRDENS', ('PlateCarree', () ,
+        ...                      {central_longitude=90, globe=None} ))
+        >>> p._setup_plots()
+        >>> p.plots['AIRDENS'].axes.set_global()
+        >>> p.plots['AIRDENS'].axes.coastlines()
+        >>> p.show()
+
+
+        This will create a RoatatedPole projection with the unrotated pole
+        position at 37.5 degrees latitude and 177.5 degrees longitude by
+        passing them in as args.
+
+
+        >>> import yt
+        >>> ds = yt.load("")
+        >>> p = yt.SlicePlot(ds, "altitude", "AIRDENS")
+        >>> p.set_mpl_projection(('RotatedPole', (177.5, 37.5))
+        >>> p._setup_plots()
+        >>> p.plots['AIRDENS'].axes.set_global()
+        >>> p.plots['AIRDENS'].axes.coastlines()
+        >>> p.show()
+
+        This will create a RoatatedPole projection with the unrotated pole
+        position at 37.5 degrees latitude and 177.5 degrees longitude by
+        passing them in as kwargs.
+
+        >>> import yt
+        >>> ds = yt.load("")
+        >>> p = yt.SlicePlot(ds, "altitude", "AIRDENS")
+        >>> p.set_mpl_projection(('RotatedPole', (), {'pole_latitude':37.5,
+        ...                       'pole_longitude':177.5}))
+        >>> p._setup_plots()
```

Comment:
Is there a reason why we have to call `_setup_plots()` directly after?

Reply:
The `set_mpl_projection()` function is decorated with two invalidate functions, one of which removes the axes from the plot. In order to access them again (and add annotations like coastlines)  _setup_plots() needs to be called. I think I noted this in the main documentation for this feature, but not the docstring.